tests/perception/developmental_primitive_perception.py

```diff
@@ -0,0 +1,65 @@
+from adam.ontology.phase1_ontology import GAILA_PHASE_1_ONTOLOGY, MOM, DAD
+from adam.perception import PerceptualRepresentation
+from adam.perception.developmental_primitive_perception import (
+    DevelopmentalPrimitivePerceptionFrame,
+    ObjectPerception,
+    SENTIENT,
+    HasFlagProperty,
+    RgbColorPerception,
+    HasColor,
+    SUPPORTS,
+    RelationPerception,
+    CONTACTS,
+    BELOW,
+    ABOVE,
+    IsRecognizedParticular,
+)
+
+
+def test_recognized_particular():
+    # create a simple situation consisting of only Mom and Dad
+    mom = ObjectPerception("mom")
+    dad = ObjectPerception("dad")
+
+    PerceptualRepresentation.single_frame(
+        DevelopmentalPrimitivePerceptionFrame(
+            perceived_objects=[mom, dad],
+            property_assertions=[
+                HasFlagProperty(mom, SENTIENT),
+                HasFlagProperty(dad, SENTIENT),
+                IsRecognizedParticular(mom, MOM, ontology=GAILA_PHASE_1_ONTOLOGY),
+                IsRecognizedParticular(dad, DAD, ontology=GAILA_PHASE_1_ONTOLOGY),
+            ],
+        )
+    )
+
+
+def test_color():
+    # create a situation with a red ball
+    red = RgbColorPerception(255, 0, 0)
+    ball = ObjectPerception("ball")
+
+    PerceptualRepresentation.single_frame(
+        DevelopmentalPrimitivePerceptionFrame(
+            perceived_objects=[ball], property_assertions=[HasColor(ball, red)]
+        )
+    )
+
+
+def test_relations():
+    # ball on a table
+    ball = ObjectPerception("ball")
+    table = ObjectPerception("table")
+
+    PerceptualRepresentation.single_frame(
+        ObjectPerception(
```

Comment:
I'm not sure if this test is correct - If I understand the structures correctly, ObjectPerception is for objects (not frames). Should the argument be DevelopmentalPrimitivePerceptionFrame(...) instead?

Reply:
Good catch. This is fixed in 92745a2